MC/config/PWGDQ/EvtGen/GeneratorEvtGen.C

```diff
@@ -106,10 +106,10 @@ class GeneratorEvtGen : public T
   {
     auto nparticles = T::mParticles.size();
     for (Int_t iparticle = 0; iparticle < nparticles; ++iparticle) {
-      auto particle = (TParticle)T::mParticles.at(iparticle);
+      const auto& particle = T::mParticles.at(iparticle);
```

Comment:
Good if it works... but I don't see where this should make a difference.

Reply:
It makes a difference only in the log messages. By using the copy the Jpsi will have no daughters before and after the decay because we're not referring to the particle in the stack. By taking a reference instead we see that as expected the Jpsi has no daughters before decay (turned off by a cfg switch), but it does after.